src/main/java/org/testng/internal/MethodInvocationHelper.java

```diff
@@ -288,7 +288,6 @@ private static void invokeWithTimeoutWithNewExecutor(ITestNGMethod tm, Object in
       ThreadTimeoutException exception = new ThreadTimeoutException("Method "
           + tm.getQualifiedName() + "()"
           + " didn't finish within the time-out " + realTimeOut);
-      exception.setStackTrace(exec.getStackTraces()[0]);

```

Comment:
Ok, thanks for the explanations.

I just don't understand why `getStackTraces` returns an empty array.
I think something is broken in `ThreadFactoryImpl` which doesn't add new threads in the list.

Could you check?